packages/config/src/class-config.php

```diff
@@ -197,4 +212,62 @@ protected function enable_connection() {
 		return true;
 	}
 
+	/**
+	 * Setup the Connection options.
+	 */
+	protected function ensure_options_connection() {
+		$options = $this->get_feature_options( 'connection' );
+
+		if ( ! empty( $options['slug'] ) ) {
+			add_action(
+				'jetpack_connection_configure_plugin',
+				function() use ( $options ) {
+					if ( empty( $options['slug'] ) ) {
+						return;
+					}
+
+					$slug = $options['slug'];
+					unset( $options['slug'] );
+
+					$name = $slug;
+					if ( ! empty( $options['name'] ) ) {
+						$name = $options['name'];
+						unset( $options['name'] );
```

Comment:
That's a good question.

The `slug` and `name` values are passed as arguments, so we remove them from the options.
Since these keys aren't in the whitelist, they will be removed either way, so that's kind of excessive. By removing them excessively the code makes it clear that these values don't go into the options.

The `Config` class doesn't have a slightest idea of what's going on in the `Plugin`, so it shouldn't rely on whether it filters out excessive data or not. The filter is the last resort, the outside code shouldn't care whether it exists or not, and only provide data that actually needs to be stored in the options.

Besides, if somebody reads the code might ask: "Why `name` and `slug` are passed twice? Should I do the same". That question shouldn't appear in their mind.

Reply:
Thanks for the explanation! What do you think of adding a short comment that explains that we're just cleaning up  `$options` before passing it to `Plugin::add()`?

Reply:
I added the comment, thanks for pointing that out.